core/src/main/java/org/pentaho/di/connections/ConnectionManager.java

```diff
@@ -282,39 +368,74 @@ public <T extends ConnectionDetails> boolean save( T connectionDetails ) {
     return success;
   }
 
+  // region test
   /**
-   * Run a test operation on the named connection
+   * Tests if a connection is valid, given its details, with default testing options.
+   * <p>
+   * If the given connection details is a VFS connection, this method delegates to {@link #test(VFSConnectionDetails)}.
+   * Otherwise, this method delegates directly to {@link ConnectionProvider#test(ConnectionDetails)}.
    *
-   * @param connectionDetails The named connection details to test
-   * @return A boolean signifying the success of the test operation
+   * @param details The details of the connection to test.
+   * @return {@code true} if the connection is valid; {@code false} otherwise.
    */
   @SuppressWarnings( "unchecked" )
-  public <T extends ConnectionDetails> boolean test( T connectionDetails ) throws KettleException {
-    ConnectionProvider<T> connectionProvider =
-      (ConnectionProvider<T>) connectionProviders.get( connectionDetails.getType() );
-    return connectionProvider.test( connectionDetails );
+  public <T extends ConnectionDetails> boolean test( @NonNull T details ) throws KettleException {
+    // FIXME: At least Catalog is VFSConnectionDetails but only ConnectionProvider!
+    // Supposedly, as a means to not be browsable by File Open Save Dialog (which only shows VFS providers).
+    // Instead, define VFSConnectionProvider#isBrowsable().
+    ConnectionProvider<T> provider = (ConnectionProvider<T>) connectionProviders.get( details.getType() );
```

Comment:
This will be addressed in a separate improvement issue, TBD. Will update this comment with issue number.